service/qos/standard_service_level_distributed_data_accessor.cc

```diff
@@ -23,12 +25,21 @@ future<qos::service_levels_info> standard_service_level_distributed_data_accesso
     return _sys_dist_ks.get_service_level(service_level_name);
 }
 
-future<> standard_service_level_distributed_data_accessor::set_service_level(sstring service_level_name, qos::service_level_options slo) const {
+future<> standard_service_level_distributed_data_accessor::set_service_level(sstring service_level_name, qos::service_level_options slo, std::optional<service::group0_guard>, abort_source*) const {
     return _sys_dist_ks.set_service_level(service_level_name, slo);
 }
 
-future<> standard_service_level_distributed_data_accessor::drop_service_level(sstring service_level_name) const {
+future<> standard_service_level_distributed_data_accessor::drop_service_level(sstring service_level_name, std::optional<service::group0_guard>, abort_source*) const {
     return _sys_dist_ks.drop_service_level(service_level_name);
 }
 
+bool standard_service_level_distributed_data_accessor::is_v2() const {
+    return false;
+}
+
+::shared_ptr<service_level_controller::service_level_distributed_data_accessor> standard_service_level_distributed_data_accessor::upgrade_to_v2(cql3::query_processor& qp, service::raft_group0_client& group0_client) const {
```

Comment:
this func is problematic, 
1. despite it's name it doesn't upgrade anything in the `standard_service_level_distributed_data_accessor` and is not static
2. I am not sure if in this case pushing the logic to the bottom of call hierarchy is best, especially why standard_service_level_distributed_data_accessor has to even know about raft_service_level_distributed_data_accessor, they are on the same "level" ?

Maybe simply having a bool/enum in the controller would be simpler? (alternative may be dynamic cast but some may disagree). 

Alternatively you can even use is_v2 func instead. But again, taking a closer look, what does it mean to be v2? In theory you can have 2 classes which declare themselves as v2, and then how to upgrade? I think what you want to express is that class A is v1, and class B is v2. Enum/bool in the controller would express it will less ambiguity.

Reply:
I agree the name should be more like `create_...`.

But I wanted to push logic of creating upgraded data accessor down to the data accessor itself because:
- I want to support `unit_test_service_levels_accessor`.
  So now `standard_service_level_distributed_data_accessor` is upgraded to `raft_service_level_distributed_data_accessor`
  and `unit_test_service_levels_accessor` is upgraded to `unit_test_raft_service_levels_accessor`
- I don't want to pull data accessor into `service_level_controller`, especially the test ones (it's not even possible), so I don't see how enum in the controller should work

Be v2 means the data accessor uses `system.service_levels_v2` table instead of `system_distributed.service_levels` (I should probably document it somewhere)